From eb56c580e27b94f3fb8645c321f86e6e2fb3cf18 Mon Sep 17 00:00:00 2001
From: Ivan Pavlov <AuthorReflex@gmail.com>
Date: Sat, 3 Oct 2026 18:28:39 +0300
Subject: [PATCH 133/138] openssl: update to 3.5.9

Is a security patch release.  The most severe CVE fixed
in this release is High.

This release incorporates the following bug fixes and mitigations:

  * Fixed DTLS retransmissions of handshake messages from a stale buffer offset.
    ([CVE-2026-84782])

  * Fixed excessive memory allocation in relative CRLDP processing.
    ([CVE-2026-35189])

  * Fixed QUIC unvalidated amplification credit may be over-accounted.
    ([CVE-2026-35191])

  * Fixed potential CPU DoS via O(n^2) fragment reassembly in QUIC.
    ([CVE-2026-42772])

  * Fixed a timing side-channel in scalar multiplication for mon-NIST EC curves.
    ([CVE-2026-54872])

  * Fixed QUIC `STREAM` fragment metadata DoS.
    ([CVE-2026-54873])

  * Fixed non-constant-time SM2 scalar multiplication on ARM64 and RISC-V.
    ([CVE-2026-54875])

  * Fixed out-of-bounds access after `SSL_set_SSL_CTX()` during a handshake.
    ([CVE-2026-72897])

  * Fixed QUIC connection-level flow control was not enforced for streams.
    ([CVE-2026-75804])

  * Fixed a NULL pointer dereference in CMP client revocation response handling.
    ([CVE-2026-75805])

  * Fixed an unauthenticated and undersized DTLS 1.2 AEAD record causing DoS.
    ([CVE-2026-75806])

  * Fixed a timing side-channel in SM2 signature generation.
    ([CVE-2026-77696])

  * Fixed an unbounded `RETIRE_CONNECTION_ID` backlog in QUIC stack
    implementation.
    ([CVE-2026-84784])

  * Fixed a bug where `EVP_DecryptFinal()` incorrectly reported a stale success
    on AES-SIV authentication failure.

Signed-off-by: Ivan Pavlov <AuthorReflex@gmail.com>

(cherry picked from commit 5edcc1c43cb97048b506168fbbe00538956796d6)
Link: https://github.com/openwrt/openwrt/pull/25558
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
---
 package/libs/openssl/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/package/libs/openssl/Makefile b/package/libs/openssl/Makefile
index 5b57fc3596..1ae14e3325 100644
--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -8,8 +8,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openssl
-PKG_VERSION:=3.5.8
-PKG_RELEASE:=2
+PKG_VERSION:=3.5.9
+PKG_RELEASE:=1
 PKG_BUILD_FLAGS:=no-mips16 gc-sections no-lto
 
 PKG_BUILD_PARALLEL:=1
@@ -21,7 +21,7 @@ PKG_SOURCE_URL:= \
 	https://www.openssl.org/source/old/$(PKG_BASE)/ \
 	https://github.com/openssl/openssl/releases/download/$(PKG_NAME)-$(PKG_VERSION)/
 
-PKG_HASH:=a8f84a39918ec6415ce765d9b429d313ba97b8143169c172e734b9514464f5b2
+PKG_HASH:=603f5602e2eef00d77fbd429d34dcd5822bb301757a1bc9cdb24c670f1eb859a
 
 PKG_LICENSE:=Apache-2.0
 PKG_LICENSE_FILES:=LICENSE.txt
-- 
2.34.1

